tests/diff.test.js

```diff
@@ -743,3 +744,59 @@ test('Should diff keyed text nodes', () => {
     expect(newNode.children.length).toBe(currentNode.children.length);
     expect(newNode.children).toEqual(currentNode.children);
 });
+
+test('Should copy simple nested DOM into VTree', () => {
+    var document = new jsdom.JSDOM().window.document;
+    var body = document.body;
+    var div = document.createElement("div");
+    body.appendChild(div);
+    var nestedDiv = document.createElement("div");
+    div.appendChild(nestedDiv);
+    var txt = document.createTextNode("foo");
+    nestedDiv.appendChild(txt);
+    var currentNode = vnodeKids('div', [ vnodeKids('div', [ vtext("foo") ]) ]);
+    copyDOMIntoVTree(currentNode, document);
+    expect(currentNode.children[0].children[0].text).toEqual('foo');
+});
+
+test('Should throw exception about expecting text node', () => {
+    var document = new jsdom.JSDOM().window.document;
+    var body = document.body;
+    var div = document.createElement("div");
+    body.appendChild(div);
+    var nestedDiv = document.createElement("div");
+    div.appendChild(nestedDiv);
+    var currentNode = vnodeKids('div', [ vtext("foo") ]);
+    const f = () => {
+        copyDOMIntoVTree(currentNode, document);
+    };
+    expect(f).toThrow(MisoException);
+});
+
+test('Should throw exception about expecting element', () => {
+    var document = new jsdom.JSDOM().window.document;
+    var body = document.body;
+    var div = document.createElement("div");
+    body.appendChild(div);
+    var txt = document.createTextNode("foo");
+    div.appendChild(txt);
+    var currentNode = vnodeKids('div', [ vnode('div', []) ]);
+    const f = () => {
+        copyDOMIntoVTree(currentNode, document);
+    };
+    expect(f).toThrow(MisoException);
+});
+
+test('Should copy DOM into VTree with multiple consecutive text nodes', () => {
+    var document = new jsdom.JSDOM().window.document;
+    var body = document.body;
+    var div = document.createElement("div");
+    body.appendChild(div);
+    var txt = document.createTextNode("foobarbaz");
+    div.appendChild(txt);
+    var currentNode = vnodeKids('div', [ vtext("foo"), vtext("bar"), vtext("baz") ]);
+    copyDOMIntoVTree(currentNode, document);
+    expect(currentNode.children[0].text).toEqual('foobarbaz');
+    expect(currentNode.children[1].text).toEqual('');
+    expect(currentNode.children[2].text).toEqual('');
```

Comment:
Shouldn’t this expect the children to each contain “foo”, “bar”, “bar”, respectively? This would be an example of the dom and vdom losing synchrony and should not be allowed.